### PR TITLE
Charlesmchen/byte parser

### PR DESCRIPTION
**Pull request by charlesmchen-signal**
PTAL @michaelkirk

### CODE REVIEW

File: Signal/test/util/ByteParserTest.swift [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+//
+
+import XCTest
+@testable import Signal
+
+class ByteParserTest: XCTestCase {
+
+    override func setUp() {
+        super.setUp()
+//         self.imageCache = ImageCache()
```
Comment (michaelkirk-signal) [inline Signal/test/util/ByteParserTest.swift]: vestigial?

File: Signal/test/util/ByteParserTest.swift [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+        XCTAssertNotNil(parser)
+        XCTAssertFalse(parser.hasError)
+
+        XCTAssertEqual(0, parser.nextShort())
+        XCTAssertTrue(parser.hasError)
+    }
+
+    func testGetShort() {
+        let data = Data(bytes: [0x01, 0x00, 0x00, 0x01, 0x01, 0x01 ])
+        let parser = ByteParser(data: data, littleEndian: true)
```
Comment (michaelkirk-signal) [inline Signal/test/util/ByteParserTest.swift]: a `littleEndian: false` test might be nice.

File: Signal/test/util/ByteParserTest.swift (hunk 1 above)
Comment (charlesmchen-signal) [reply Signal/test/util/ByteParserTest.swift]: Removed.

File: Signal/test/util/ByteParserTest.swift (hunk 2 above)
Comment (charlesmchen-signal) [reply Signal/test/util/ByteParserTest.swift]: Done.

**Pull request merged**